### ISSUE TITLE
Deploy fleet repo to downstream clusters when in Airgap

### ISSUE THREAD

**Issue by slickwarren** [rancher/qa-tasks#1524]

**Comment by slickwarren** [comment]
attempt 1: deploy a git service in a container to use in airgap, copy files to it that are accessible to the new service, use that as an airgapped git repo for fleet to access
* using gitea in a single pod seems the easiest for this compared to something like gitlab, but the problems I ran into were:
  * copying files to the 'expected' dir still doesn't treat it like a git repo. gitea has to process it first as an 'unadopted repository', which you need to login as an admin -> run as this user to fix. this is probably the best approach I found but it would require doing something like some http calls to do this through its UI, so not the best solution
  * using gitlab was basically the same problem, but setup for gitlab was more complicated than gitlab and took more resources

attempt 2: use a helm chart 'directly' instead -> this isn't an option for fleet, though they do have a 'helmRepo' option, that helm repo still has to point to a valid gitRepo. 

attempt 3: publish the files a different way for fleet to use
  * nginx as a fileserver didn't publish the repo.git in a way that fleet expected / liked, so this isn't viable option
  * the cli tool git suggests you can use localpaths as a git repo. This actually did work (copying the files to gitJob container) however since fleet isn't built to take in local paths, the subsequent job(s) failed since they couldn't access the volumes. I was using hostVol and emptyDir in gitJob. Since the gitjob-controller is written to create other jobs that actually do the git cloning, those jobs don't have a way to pass along volumes nor create new ones through the API. So this wouldn't work from what I could tell. 


**tldr; the most viable option would require:** 
* installing gitea with a mounted volume that has the git repo copied from the bastion node, avoiding its API since that would require creating a token manually 
  * (or perhaps there's a hacky way to do it like we do in v3, where you can get a temporary token but it would require digging), 
* publish the service so that it can be accessed 
* logging in as the administrator, 
* add the unadopted repository from the newly mounted volume
* get the http git repo URL and use that for fleet's gitRepo
